packages/util/src/object.ts

```diff
@@ -108,24 +110,103 @@ export const assoc = <T>(key: string, value: any, obj: T) => ({
 });
 
 /**
- * Flattens a array: [1,[2,[3,4]],5] will become [1,2,3,4,5].
- * Non primitive values are copied by reference.
- *
- * @param input - array to be flattened
- * @returns the flat array.
+ * Create an object based on the given key and value if neither key nor value are nullish.
+ * @param key - Name of the header.
+ * @param value - Value of the header.
+ * @returns - An object containing the given key and value of an empty object.
  */
-export const flatten = (input: any[]): any[] => {
-  const flatResult: any[] = [];
-  const stack: any[] = [...input];
+export function toSanitizedObject(
+  key: string,
+  value: any
+): Record<string, any> {
+  return isNullish(key) || isNullish(value) ? {} : { [key]: value };
+}
 
-  while (stack.length > 0) {
-    const current = stack.pop();
-    if (!Array.isArray(current)) {
-      flatResult.push(current);
-    } else {
-      stack.push(...current);
-    }
-  }
+/**
+ * Pick a partial object from the given object indepdendent of casing - if available.
```

Comment:
mention for all the pick methods that a reference on the found object is returned. I think pick is descriptive there but does not hurt to be extra explicit in the docs.

Reply:
Done.